lib/rx/client.rb

```diff
@@ -189,6 +199,13 @@ def cache_key(action)
       "#{session.user_id}:#{action}"
     end
 
+    def increment_refill(count = 1)
```

Comment:
Is there a case where `@upstream_request` will be nil? I can't think of a reason it would ever be nil coming from the controller, so the only case I can think of is if someone manually passed nil when instantiating the class. I think it may make more sense to default to nil in the initializer, because if the request is ever an empty hash, it will error on `@upstream_request.env['SOURCE_APP']`. You could even return early if the request is nil, although that may more effectively hide the fact that something is wrong.

Also, does returning nil at the end of the method serve any purpose? It doesn't seem like we're relying on the return value in any way.

Reply:
Thank you, really good feedback and your suggestions make sense, implemented 🙂